### PR TITLE
 #222 -  Add contact share button

### LINKED ISSUE #222: Share your QR code
There should be a feature for the user to share his/her QR code with others on different social media apps

### PR DESCRIPTION
**Pull request by f71uday**
# Description

added menu for displaying share button
added functionality onClick share menu

Fixes #222 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Swati4star** [comment]
Is it working fine? 
@f71uday An image with QR code is shared?

File: Android/app/src/main/res/menu/share_contact_menu.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/share_contact"
+        android:title="share"
```
Comment (Swati4star) [inline Android/app/src/main/res/menu/share_contact_menu.xml]: add "share" string to strings.xml file, and use that here.
Don't add constant strings directly in xml file

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java [hunk 2]
```diff
@@ -151,4 +164,32 @@ public static Intent getStartIntent(Context context) {
...
+        }
+        Uri uri = FileProvider.getUriForFile(Objects.requireNonNull(this) ,
+                "io.github.project_travel_mate.shareFile" , mFile);
+        Intent intent = new Intent();
+        intent.setAction(Intent.ACTION_SEND);
+        intent.setType("image/*");
+        intent.putExtra(Intent.EXTRA_SUBJECT, "");
+        intent.putExtra(Intent.EXTRA_TEXT , getString(R.string.share_contact_qr));
+        intent.putExtra(Intent.EXTRA_STREAM , uri);
+        startActivity(Intent.createChooser(intent , "Share Contact Details"));
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: Add ` "Share Contact Details"` also in strings.xml file & use that here

**Comment by f71uday** [comment]
yes it is working absolutely fine

**Comment by Swati4star** [comment]
Awesome! Just incorporate the changes I have mentioned.

**Comment by f71uday** [comment]
screenshot_20180714-202645
@Swati4star demo is here

File: Android/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -173,6 +176,6 @@
     <string name="no_notifications">You don\'t have any notifications.</string>
     <string name="mark_all_read_notifications">Do you want to mark all notifications as read?</string>
 
-    <string name="share_fun_fact">Hey! Checkout this fun fact, seems interesting! I find this on Travel Mate app. Download it from plastore :
+    <string name="share_fun_fact">Hey! Checkout this fun fact, seems interesting! I find this on Travel Mate app. Download it from Play Store :
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: thanks for correcting this typo :)

Comment (f71uday) [reply Android/app/src/main/res/values/strings.xml]: Hehe!!

**Pull request merged**